avm/res/compute/gallery/image/main.bicep

```
@@ -12,108 +14,69 @@ param location string = resourceGroup().location
-- If this value is not specified, then it is determined by the securityType parameter.
-- If the securityType parameter is specified, then the value of hyperVGeneration will be V2, else V1.
-''')
-@allowed([
-  'V1'
-  'V2'
-])
-param hyperVGeneration string?
+@sys.description('Required. This is the gallery image definition identifier.')
+param identifier identifierType
 
-@sys.description('Optional. The security type of the image. Requires a hyperVGeneration V2.')
-@allowed([
-  'Standard'
-  'TrustedLaunch'
-  'ConfidentialVM'
-  'ConfidentialVMSupported'
-])
-param securityType string = 'Standard'
+@sys.description('Required. This property allows the user to specify the state of the OS of the image.')
+param osState ('Generalized' | 'Specialized')
 
-@sys.description('Optional. Specifiy if the image supports hibernation.')
-param isHibernateSupported bool = false
+@sys.description('Required. This property allows you to specify the type of the OS that is included in the disk when creating a VM from a managed image.')
+param osType ('Linux' | 'Windows')
 
-@sys.description('''Optional. Specify if the image supports accelerated networking.
-Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, greatly improving its networking performance.
-This high-performance path bypasses the host from the data path, which reduces latency, jitter, and CPU utilization for the most demanding network workloads on supported VM types.
-''')
-param isAcceleratedNetworkSupported bool = true
+@sys.description('Optional. The privacy statement uri.')
+param privacyStatementUri string?
 
-@sys.description('Optional. The description of this gallery Image Definition resource. This property is updatable.')
-param description string?
+@sys.description('Optional. Describes the gallery image definition purchase plan. This is used by marketplace images.')
+param purchasePlan purchasePlanType?
 
-@sys.description('Optional. The Eula agreement for the gallery Image Definition. Has to be a valid URL.')
-param eula string?
+@sys.description('Optional. Describes the resource range (1-128 CPU cores). Defaults to min=1, max=4.')
+param vCPUs resourceRangeType?
 
-@sys.description('Optional. The privacy statement uri. Has to be a valid URL.')
-param privacyStatementUri string?
+@sys.description('Optional. Describes the resource range (1-4000 GB RAM). Defaults to min=4, max=16.')
+param memory resourceRangeType?
 
 @sys.description('Optional. The release note uri. Has to be a valid URL.')
 param releaseNoteUri string?
 
-@sys.description('Optional. The product ID.')
-param productName string?
+@sys.description('Optional. The security type of the image. Requires a hyperVGeneration V2.')
+param securityType ('Standard' | 'TrustedLaunch' | 'ConfidentialVM' | 'ConfidentialVMSupported')?
+
+@sys.description('Optional. Specify if the image supports accelerated networking. Defaults to true.')
+param isAcceleratedNetworkSupported bool?
```

```suggestion
@sys.description('Optional. Specify if the image supports accelerated networking.')
param isAcceleratedNetworkSupported bool = true
```


I'd suggest to check this for other parameters too. With it now being a parameter, we can have a default and hence can set the default directly 💪